src/Core/src/Platform/Android/MauiContext.cs

```diff
@@ -19,26 +25,78 @@ public MauiContext(IServiceProvider services, Context context) : this(context)
 		{
 			_services = services ?? throw new ArgumentNullException(nameof(services));
 			_mauiHandlersServiceProvider = Services.GetRequiredService<IMauiHandlersServiceProvider>();
+			_targetIdiom = TargetIdiom.Unsupported;
 		}
 
-		public Context? Context
+		public Context? Context =>
+			_context.TryGetTarget(out Context? context) ? context : null;
+
+		public IServiceProvider Services =>
+			_services ?? throw new InvalidOperationException($"No service provider was specified during construction.");
+
+		public IMauiHandlersServiceProvider Handlers =>
+			_mauiHandlersServiceProvider ?? throw new InvalidOperationException($"No service provider was specified during construction.");
+
+		public TargetIdiom Idiom
```

Comment:
Does this really need to be reimplemented? Xamarin.Essentials already have an implementation for it. https://docs.microsoft.com/en-us/xamarin/essentials/device-information?tabs=ios

Reply:
Here https://github.com/xamarin/Essentials/blob/main/Xamarin.Essentials/DeviceInfo/DeviceInfo.android.cs#L31-L73

Reply:
You are right, must use Essentials as specified in Spec https://github.com/dotnet/maui/issues/1305